azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -71,6 +93,7 @@
     interface Definition<ParentT> extends
         DefinitionStages.Blank<ParentT>,
         DefinitionStages.WithAddressPrefix<ParentT>,
+        DefinitionStages.WithNetworkSecurityGroup<ParentT>,
         DefinitionStages.WithAttach<ParentT> {
     }
 
```

Comment:
...so thinking about this a little more, here's I think the key differences between the two approaches that emerge in the following scenarios:
A. imagine we are changing a required wither to be optional. 
  i. in the approach i'm following, these steps need to be taken:
    1. change the wither's interface to have a different return value (Attachable/Creatable)
    2. add the wither's interface to the Attachable/Creatable interface "extends" list
ii. the approach you're following, these steps would be needed:
    1. change the wither's interface to have a different return value (Attachable/Creatable)
    2. add the wither's interface to the Attachable/Creatable interface "extends" list
    3. remove the wither's interface from the Definition "extends" list

There are more steps in the latter approach, so harder to remember/easier to make mistakes. Similarly for going from optional to required.

B. However, there is also a scenario where the advantages are flipped -- adding a new wither. In that case, My approach requires the wither to be added to both the Definition and the WithAttach, in your approach i only need to add it to WithAttach. So here, making that kidn of change is easier in your approach.

So it;'s kind of a wash in that regard.

Need to think some more about this... :-)